packages/widget-chat/src/actions/flags.js

```diff
@@ -0,0 +1,98 @@
+export const ADD_FLAG = `ADD_FLAG`;
+function actionAddFlag(activity) {
+  return {
+    type: ADD_FLAG,
+    activity
+  };
+}
+
+export const BEGIN_RECEIVE_FLAGS = `BEGIN_RECEIVE_FLAGS`;
+function actionBeginReceiveFlags() {
+  return {
+    type: BEGIN_RECEIVE_FLAGS
+  };
+}
+
+export const RECEIVE_FLAGS = `RECEIVE_FLAGS`;
+function actionReceiveFlags(flags) {
+  return {
+    type: RECEIVE_FLAGS,
+    flags
+  };
+}
+
+export const REMOVE_FLAG = `REMOVE_FLAG`;
+function actionRemoveFlag(flag) {
+  return {
+    type: REMOVE_FLAG,
+    flag
+  };
+}
+
+export const REMOVE_FLAG_FAIL = `REMOVE_FLAG_FAIL`;
+function actionRemoveFlagFail(flag) {
+  return {
+    type: REMOVE_FLAG_FAIL,
+    flag
+  };
+}
+
+export const UPDATE_FLAG = `UPDATE_FLAG`;
+function actionUpdateFlag(flag) {
+  return {
+    type: UPDATE_FLAG,
+    flag
+  };
+}
+
+/**
+ * Fetches all of the current user's flags
+ *
+ * @param {any} spark
+ * @returns {function}
+ */
```

Comment:
Nice job on the jsdocs... makes me feel like I should add them to my stuff as well.